pkg/webhook/vlanconfig/validator.go

```diff
@@ -92,29 +92,35 @@ func (v *Validator) Update(_ *types.Request, oldObj, newObj runtime.Object) erro
 		return fmt.Errorf(updateErr, newVc.Name, err)
 	}
 
-	// get affected nodes after updating
 	oldNodes, err := getMatchNodesMap(oldVc)
 	if err != nil {
 		return fmt.Errorf(updateErr, oldVc.Name, err)
 	}
-	var deltaNodes map[string]bool
-	if oldVc.Spec.ClusterNetwork != newVc.Spec.ClusterNetwork {
-		deltaNodes = oldNodes
-	} else {
-		for n := range oldNodes {
-			if !newNodes[n] {
-				deltaNodes[n] = true
-			}
-		}
-	}
 
-	if err := v.checkVmi(oldVc, deltaNodes); err != nil {
+	// get affected nodes after updating
+	affectedNodes := getAffectedNodesMap(oldVc.Spec.ClusterNetwork, newVc.Spec.ClusterNetwork, oldNodes, newNodes)
+	if err := v.checkVmi(oldVc, affectedNodes); err != nil {
 		return fmt.Errorf(updateErr, oldVc.Name, err)
 	}
 
 	return nil
 }
 
+func getAffectedNodesMap(oldCn, newCn string, oldNodesMap, newNodesMap map[string]bool) map[string]bool {
+	affectedNodesMap := make(map[string]bool, len(oldNodesMap))
+	if newCn != oldCn {
+		affectedNodesMap = oldNodesMap
+	} else {
+		for n := range oldNodesMap {
+			if !newNodesMap[n] {
+				affectedNodesMap[n] = true
+			}
+		}
```

Comment:
Consider using github.com/deckarep/golang-set to find the difference set, it will be more simple.
```go
var set1 = mapset.NewSet[string]()
	set1.Add("a")
	set1.Add("b")

	var set2 = mapset.NewSet[string]()
	set2.Add("c")

	var set3 = set1.Difference(set2)
	fmt.Println(set3)
```